### PR TITLE
Update participation response

### LINKED ISSUE #7118: eth/v1alpha1/validators/participation to include source and head %
From feedbacks, several users have reported that granularity of seeing head vs target vs source has been useful when following other client's metrics such as LH.  Each "participation rate" has its own meaning about network performance, it'll be great if Prysm can adopt to this standard as well

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**
Implements https://github.com/prysmaticlabs/ethereumapis/pull/175 

**Which issues(s) does this PR fix?**

Fixes #7118

**Other notes for review**
```
➜  prysm git:(participation) ✗ curl -X GET "http://localhost:3051/eth/v1alpha1/validators/participation?epoch=4805" -H "accept: application/json"
{"epoch":"4805","finalized":true,"participation":{"globalParticipationRate":0.779474,"votedEther":"801272000000000","eligibleEther":"1027965000000000","currentEpochActiveGwei":"1028093000000000","currentEpochAttestingGwei":"757813000000000","currentEpochTargetAttestingGwei":"757443000000000","previousEpochActiveGwei":"1027965000000000","previousEpochAttestingGwei":"807622000000000","previousEpochTargetAttestingGwei":"801272000000000","previousEpochHeadAttestingGwei":"413175000000000"}}%
```

### CODE REVIEW

File: beacon-chain/rpc/validator/exit.go [hunk 1]
```diff
@@ -15,7 +14,7 @@ import (
 )
 
 // ProposeExit proposes an exit for a validator.
-func (vs *Server) ProposeExit(ctx context.Context, req *ethpb.SignedVoluntaryExit) (*ptypes.Empty, error) {
+func (vs *Server) ProposeExit(ctx context.Context, req *ethpb.SignedVoluntaryExit) (*ethpb.ProposeExitResponse, error) {
```
Comment (terencechain) [inline beacon-chain/rpc/validator/exit.go:17]: Note. I had to update this to comply with latest ethapi commit. #7125 was going to address this

File: beacon-chain/rpc/beacon/validators.go [hunk 2]
```diff
@@ -485,7 +485,7 @@ func (bs *Server) GetValidatorParticipation(
 		return nil, status.Error(codes.Internal, "Could not get state")
 	}
 
-	requestedState, err = bs.appendNonFinalizedBlockAttsToState(ctx, requestedState, requestedEpoch)
+	//requestedState, err = bs.appendNonFinalizedBlockAttsToState(ctx, requestedState, requestedEpoch)
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beacon/validators.go:488]: Please address this

File: beacon-chain/rpc/beacon/validators.go [hunk 3]
```diff
@@ -504,9 +504,16 @@ func (bs *Server) GetValidatorParticipation(
 		Epoch:     requestedEpoch,
 		Finalized: requestedEpoch <= headState.FinalizedCheckpointEpoch(),
 		Participation: &ethpb.ValidatorParticipation{
-			GlobalParticipationRate: float32(b.PrevEpochTargetAttested) / float32(b.ActivePrevEpoch),
-			VotedEther:              b.PrevEpochTargetAttested,
-			EligibleEther:           b.ActivePrevEpoch,
+			GlobalParticipationRate:          float32(b.PrevEpochTargetAttested) / float32(b.ActivePrevEpoch),
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beacon/validators.go:507]: Please comment that these three fields are deprecated and should be removed soon. Also add tracking issue to remove so we don't forget.

File: beacon-chain/rpc/beacon/validators.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/rpc/beacon/validators.go:488]: oops sorry, i had a bad merge

**Pull request merged**